tests/scripts/all.sh

```diff
@@ -1290,6 +1290,38 @@ component_test_no_use_psa_crypto_full_cmake_asan() {
     if_build_succeeded env OPENSSL_CMD="$OPENSSL_NEXT" tests/compat.sh -e '^$' -f 'ARIA\|CHACHA'
 }
 
+component_test_psa_crypto_config_basic() {
+    # full plus MBEDTLS_PSA_CRYPTO_CONFIG
+    msg "build: full + MBEDTLS_PSA_CRYPTO_CONFIG"
+    scripts/config.py full
+    scripts/config.py set MBEDTLS_PSA_CRYPTO_CONFIG
+    scripts/config.py set MBEDTLS_PSA_CRYPTO_DRIVERS
+    scripts/config.py unset MBEDTLS_USE_PSA_CRYPTO
+    # Need to define the correct symbol and include the test driver header path in order to build with the test driver
+    make CC=gcc CFLAGS="$ASAN_CFLAGS -DPSA_CRYPTO_DRIVER_TEST -I../tests/include -O2" LDFLAGS="$ASAN_CFLAGS"
+
+    msg "test: psa crypto config basic"
+    make test
+}
+
+component_test_psa_want_ecdsa_disabled_software() {
+    # full plus MBEDTLS_PSA_CRYPTO_CONFIG with PSA_WANT_ALG_ECDSA
+    # without MBEDTLS_ECDSA_C
+    msg "build: full + MBEDTLS_PSA_CRYPTO_CONFIG + PSA_WANT_ALG_ECDSA without MBEDTLS_ECDSA_C"
+    scripts/config.py full
+    scripts/config.py set MBEDTLS_PSA_CRYPTO_CONFIG
+    scripts/config.py set MBEDTLS_PSA_CRYPTO_DRIVERS
+    scripts/config.py unset MBEDTLS_USE_PSA_CRYPTO
+    scripts/config.py unset MBEDTLS_ECDSA_C
+    scripts/config.py -f include/psa/crypto_config.h set PSA_WANT_ALG_ECDSA
+    scripts/config.py -f include/psa/crypto_config.h set PSA_WANT_ALG_DETERMINISTIC_ECDSA
```

Comment:
With all those configuration settings we end up with:
PSA_WANT_ALG_ECDSA defined
MBEDTLS_PSA_ACCEL_ALG_ECDSA NOT defined
MBEDTLS_PSA_BUILTIN_ALG_ECDSA defined
MBEDTLS_ECDSA_C defined
PSA_WANT_ALG_DETERMINISTIC_ECDSA defined
MBEDTLS_PSA_ACCEL_ALG_DETERMINISTIC_ECDSA NOT defined
MBEDTLS_PSA_BUILTIN_ALG_DETERMINISTIC_ECDSA defined
MBEDTLS_ECDSA_DETERMINISTIC defined

For  MBEDTLS_PSA_BUILTIN_ALG_ECDSA not to be defined (my understanding of the purpose of the test), MBEDTLS_PSA_ACCEL_ALG_ECDSA needs to be defined.

Otherwise, I think we need to keep MBEDTLS_ECDSA_C defined as the related code is needed by the signature test driver.

Reply:
@gilles-peskine-arm Can you comment on the logic of the test here since you created the original ticket and had asked for a specific test case that I tried to capture with this PR?

Reply:
The point of this component is to build the library without `MBEDTLS_ECDSA_C`, but with `PSA_WANT_ALG_ECDSA`. This requires `MBEDTLS_PSA_CRYPTO_CONFIG` and an ECDSA accelerator driver.

Reply:
As it is the configuration is not aligned with the ticket. You unset MBEDTLS_ECDSA_C but as PSA_WANT_ALG_ECDSA  is defined and MBEDTLS_PSA_ACCEL_ALG_ECDSA is not defined, MBEDTLS_ECDSA_C is eventually defined because of the following code in config_psa.h:
```c
#if defined(PSA_WANT_ALG_ECDSA)
#if !defined(MBEDTLS_PSA_ACCEL_ALG_ECDSA)
#define MBEDTLS_PSA_BUILTIN_ALG_ECDSA                   1
#define MBEDTLS_ECDSA_C
#endif /* !MBEDTLS_PSA_ACCEL_ALG_ECDSA */
#endif /* PSA_WANT_ALG_ECDSA */
```

Reply:
@ronald-cron-arm Thanks for clarifying. If I removed the following items for the test would it complete the requirement for removing all items that rely on MBEDTLS_ECDSA_C?

MBEDTLS_KEY_EXCHANGE_ECDH_ECDSA_ENABLED
MBEDTLS_KEY_EXCHANGE_ECDHE_ECDSA_ENABLED

I just searched for the items that required MBEDTLS_ECDSA_C.

For the MBEDTLS_PSA_ACCEL_ALG_ECDSA could follow your suggestion or I could also add in all.sh like this:

make CC=gcc CFLAGS="$ASAN_CFLAGS -DPSA_CRYPTO_DRIVER_TEST -DMBEDTLS_PSA_ACCEL_ALG_ECDSA -I../tests/include -O2" LDFLAGS="$ASAN_CFLAGS"

Is one method preferred over the other? I think both accomplish the same thing by temporarily including the macro for this test.

Reply:
> If I removed the following items for the test would it complete the requirement for removing all items that rely on MBEDTLS_ECDSA_C?
MBEDTLS_KEY_EXCHANGE_ECDH_ECDSA_ENABLED
MBEDTLS_KEY_EXCHANGE_ECDHE_ECDSA_ENABLED

I don't know this on the top of my head, the point is that at the end it should compile.

Reply:
> -DMBEDTLS_PSA_ACCEL_ALG_ECDSA
I am fine with this way of doing it.

Reply:
> If I removed the following items for the test would it complete the requirement for removing all items that rely on MBEDTLS_ECDSA_C?

For public-key algorithms, the answer is already present in `tests/scripts/depends-pkalgs.pl`, and what you posted here is correct.

Reply:
This comment has been addressed in the latest version.